mllib/src/main/scala/org/apache/spark/mllib/tree/configuration/Strategy.scala

```diff
@@ -178,14 +178,14 @@ object Strategy {
    * @param algo  "Classification" or "Regression"
    */
   def defaultStrategy(algo: String): Strategy = {
-    defaultStategy(Algo.fromString(algo))
+    defaultStrategy(Algo.fromString(algo))
   }
 
   /**
    * Construct a default set of parameters for [[org.apache.spark.mllib.tree.DecisionTree]]
    * @param algo Algo.Classification or Algo.Regression
    */
-  def defaultStategy(algo: Algo): Strategy = algo match {
+  def defaultStrategy(algo: Algo): Strategy = algo match {
```

Comment:
Good point, this has been around since 1.2 so it might break backwards compatibility. Let's keep the discussion going